### PR TITLE
Session Management & Logout

### PR DESCRIPTION
**Pull request by seromarin**
## Summary
- The user session is stored in the localStorage
- Session stored:
    - access-token
    - client
    - uid
- The session is deleted on logout

## Screenshot
Captura de Pantalla 2020-11-18 a la(s) 3 12 42 p  m

## Trello Card
https://trello.com/c/j6YlFqgY/16-session-management-logout

### CODE REVIEW

File: wolox-books/src/app/screens/login/login.component.ts [hunk 1]
```diff
@@ -19,28 +20,33 @@ export class LoginComponent {
...
-      const keys = respose.headers.keys();
-      const headers = keys.map(key => `${key}: ${respose.headers.get(key)}`);
-      console.log('access-token', respose.headers.get('access-token'))
-      this.router.navigate(['/'])
+      this.persistSession(respose);
+      this.router.navigate(['app', 'list']);
     });
   }
 
+  persistSession(userSession: HttpResponse<IUserHTTPResponse>): void {
```
Comment (mcallegari10) [inline wolox-books/src/app/screens/login/login.component.ts]: I think this should be in the user service

File: wolox-books/src/app/app-routing.module.ts [hunk 2]
```diff
@@ -1,12 +1,40 @@
...
+    component: LoginComponent
+  },
+  {
+    path: 'sign-up',
+    component: RegisterComponent
+  },
+  {
+    path: 'app',
+    component: AppShellComponent,
+    // outlet: 'app-data',
```
Comment (idcuadrowolox) [inline wolox-books/src/app/app-routing.module.ts]: remove comment

File: wolox-books/src/app/screens/register/register.component.ts [hunk 3]
```diff
@@ -21,39 +21,39 @@ export class RegisterComponent{
...
+      locale: [ 'es' ]
+    });
   }
 
-  registerUser(user: IUserComplete) {
+  registerUser(user: IUserComplete): void {
     this.userService.createUser(user).subscribe((respose: IUserComplete) => {
-      console.log('success', respose)
-      this.router.navigate(['login'])
+      console.log('success', respose);
```
Comment (idcuadrowolox) [inline wolox-books/src/app/screens/register/register.component.ts]: remove console

File: wolox-books/src/app/services/local-storage.service.ts [hunk 4]
```diff
@@ -0,0 +1,20 @@
+import { Injectable } from '@angular/core';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class LocalStorageService {
+
```
Comment (idcuadrowolox) [inline wolox-books/src/app/services/local-storage.service.ts:7]: remove empty space

File: wolox-books/src/app/screens/register/register.component.ts (hunk 3 above)
Comment (seromarin) [reply wolox-books/src/app/screens/register/register.component.ts]: Removed

File: wolox-books/src/app/services/local-storage.service.ts (hunk 4 above)
Comment (seromarin) [reply wolox-books/src/app/services/local-storage.service.ts:7]: Removed

File: wolox-books/src/app/app-routing.module.ts (hunk 2 above)
Comment (seromarin) [reply wolox-books/src/app/app-routing.module.ts]: Removed

**Review by idcuadrowolox — APPROVED** [review]

**Review by mcallegari10 — APPROVED** [review]
Just this comment 😄

File: wolox-books/src/app/screens/login/login.component.ts (hunk 1 above)
Comment (seromarin) [reply wolox-books/src/app/screens/login/login.component.ts]: Done

**Pull request merged**